### DISCUSSION TITLE
How to use `unique` inside an schema?

### DISCUSSION THREAD

**Discussion by aralroca** [Josuto/node-abstract-repository#219] [General]
We realize that the plugin `uniqueValidator` is not executed:

```tsx
import { SUPPORTED_LANGUAGES } from '@config/supported-languages';
import uniqueValidator from 'mongoose-unique-validator';
import { BaseSchema, extendSchema } from 'monguito';

export const SomeSchema = extendSchema(
  BaseSchema,
  {
    title: Object.fromEntries(
      SUPPORTED_LANGUAGES.map((langData) => [langData.id, { type: String, required: false }])
    ),
    externalId: { 
      type: String, 
      required: true,
      // This is not working 
      unique: true
    },
  },
  { timestamps: true }
);

// This is not working
SomeSchema.plugin(uniqueValidator);
```

When calling the `save` function of Monguito it is saved even if the `externalId` already exists. I do not know if it is a problem when using the extendSchema no longer allows plugins like `mongoose-unique-validator` or if it is something else the problem. Any idea?

**Comment by Josuto** [comment]
Many thanks for starting this discussion, @aralroca! It seems that you found a bug in Mongoose schema definition 😅. I am creating an issue to fix it. Thanks again for letting me know! 😊
